### ISSUE TITLE
[Question] BlazorStrap vs using Bootstrap directly

### ISSUE THREAD

**Issue by lonix1** [chanan/BlazorStrap#548]
Hi and thanks for making this cool library. I'm a long time user of Bootstrap, so was pleased to find this!

Newbie question - and I'm not trolling, just curious:

What is the benefit of using this library rather than just Bootstrap directly? I've looked through the docs site, and noticed that for every bootstrap concept, there is an equivalent in blazorstrap. Are there things that are easier/better when using blazorstrap?

I've looked at some other libraries (e.g. mudblazor) too, and saw that they went in a different direction (abstracting the css+js completely). Where does this library fit in that continuum?

**Comment by jbomhold3** [comment]
To put it as simple as possible. Bootstraps javascript library changes the DOM directly. These changes are not tracked by blazor, so it's not stateful. Blazor will rerender the component to what it knows about. Any changes that javascript made will be overwritten if the affected area falls in a tracked area by Blazor.  BlazorStrap and alike libraries reimplement what's done in javascript into a stateful way of doing things. So even if your component completely rerenders, the changes made are tracked and will redraw accordingly. This is also true for the stateful javascript library as well(the exact term isn't coming to mind at 5am). Like Angular or Vue.js this is why you will see a library for Bootstrap for them both as well. While it will work just to use the normal library, anytime, Blazor changes state and rerenders, all affected DOM changes will be overwritten because it has zero clue they were made. 

For example, you have a modal default closed inside a component and open it via just the normal js library. If that component calls an update, your modal will return to the closed state because Blazor has no clue the show class was added.

To sum it up will the normal library work. Sure, if you are fine with losing the changes to the DOM every time your page gets rendered.

**Comment by jbomhold3** [comment]
The other part of the question BlazorStrap is a Bootstrap focused library. We mimic the default behavior of Bootstrap so there should be little to no difference between how BlazorStrap and native Bootstrap behaves. BlazorStrap, MudBlazor, MatBlazor, Blazorise, and all the other libraries made by community members are free and you should pick the one that best suits your needs.
